src/transactions/AllowTrustOpFrame.cpp

```diff
@@ -136,6 +137,13 @@ AllowTrustOpFrame::doCheckValid(uint32_t ledgerVersion)
         innerResult().code(ALLOW_TRUST_MALFORMED);
         return false;
     }
+
+    if (!trustLineFlagIsValid(mAllowTrust.authorize))
```

Comment:
Transactions subsystem shouldn't have dependencies in the invariants subsystem. The transaction subsystem is protocol-level, whereas the invariants subsystem isn't.

Reply:
This method should be shared between the transaction and invariants subsystem. What do you think about renaming `InvariantUtils.h/cpp` and moving it to `src/util`?